mempool/v1/mempool_test.go

```diff
@@ -661,7 +661,7 @@ func TestMalleatedTxRemoval(t *testing.T) {
 	originalHash := sha256.Sum256(originalTx)
 
 	// create the wrapped child transaction
-	wTx, err := types.WrapMalleatedTx(originalHash[:], malleatedTx)
+	wTx, err := types.WrapMalleatedTx(originalHash[:], 0, malleatedTx)
```

Comment:
Question to understand better: 
How do we know that the malleated transaction will be at share 0? 
Is it because of the malleated will  be separated to a transaction and a message? the tx will be at the begining with `TxNamespaceID` and the message with another namespace?

Reply:
Tendermint should never have to interpret a share index as that is application specific logic, so in this case, we are just using a filler value of zero. you are correct that a message cannot actually have a share index of 0, the lowest possible is 1

Reply:
the specs do a really good job of outlining the necessary functionality and rationale if you haven't already read them. https://github.com/celestiaorg/celestia-specs/blob/master/src/rationale/message_block_layout.md#message-layout

Will definitely try to outline this and the application logic that actually uses this in the ADR tho